Configurability and use the Host header

mmn opened the pull request.

Commit:
Allow custom hostname (trust the Host header)

Commit:
Configurable server port with commandline option

Commit:
xattrs variable was incorrectly named

Commit:
Merge branch 'master' into transcode

Conflicts:
	server.py

options.port was added unknowingly to the master branch!

Commit:
compatibility with both xattr modules

pyxattr is in Debian's apt repositories, but does not support xattr.xattr(...)

Commit:
user.mime_type is specified by freedesktop.org for xattr

http://standards.freedesktop.org/shared-mime-info-spec/latest/ar01s02.html#idm140622087713936

Commit:
More exact MIME type string matching

Commit:
ignore files directory

Commit:
Configurable transcoder (for us with avconv for example)

ghost commented:
...a bunch of other stuff has also been added now and I'm too lazy to change the subject.

Commit:
Don't allow arbitrary filenames in arbitrary paths

Commit:
aac is experimental and needs an extra CLI option

Interestingly enough I noticed this when uploading an .mp4 from
my Jolla which the MIME magic interprets as video/quicktime.

ghost commented:
(och egentligen ska jag ju ha en egen branch som jag ber om merge ifrån, men det tar vi nästa gång)

ghost commented:
Notera även icke-bakåtkompatibiliteten för user.mime_type :)
